### ISSUE TITLE
memory leak in validation

### ISSUE THREAD

**Issue by minrk** [Julian/jsonschema#237]
[this gist](https://gist.github.com/minrk/a72d0291702d4fca1b66) includes a large JSON file and script that opens and validates it repeatedly. Each validation appears to permanently increase the memory footprint, at least on Python 3.4.

To run the test:

```bash
git clone https://gist.github.com/a72d0291702d4fca1b66.git test-leak
cd test-leak
python3 test-schema.py
```

The script loads a JSON file a few times, showing that only the first load increases the memory footprint. Then it goes through the same loop, also validating the notebook on each iteration, showing that memory increases on each validation:

```
jsonschema-2.4.0
pid: 90109
    baseline     20MB
      opened    127MB
      opened    127MB
      opened    127MB
      opened    127MB
      opened    127MB
   validated    235MB
   validated    397MB
   validated    558MB
   validated    720MB
   validated    882MB
```

**Comment by minrk** [comment]
My apologies. It looks like this is fixed in current jsonschema-2.5.1. Thanks for fixing it!

**Comment by Julian** [comment]
:) glad it works.
